CONTRIBUTING.md

```diff
@@ -29,6 +29,14 @@ cabal v2-build Cabal-tests:unit-tests # build Cabal's unit test suite
 cabal v2-build cabal-tests # etc...
 ```
 
+Building Cabal for hacking with Nix
+--------------------------
+
+```
+nix develop # if you don't have direnv installed
+cabal build cabal
+```
+
```

Comment:
I'd add something like "Nix flake is not officially supported and as such may fall out of date. Be prepared to tweak it and please contribute fixes back" to make it less controversial and burdensome.